test/es6/DeferParseLambda.js

```diff
@@ -110,6 +110,38 @@ var tests = [
             `);
         }
     },
+    {
+        name: "Global functions using 'yield' as identifier",
+        body: function () {
+            WScript.LoadScript(`
+                var a = async (yield) => { yield };

```

Comment:
What is the correct interpretation of this function? Does the `yield` parameter completely shadow the ability to `yield` as a keyword, or is the formal parameter unused here and the function would just `yield` undefined when called?

Reply:
This is not a generator so `yield` is treated as a normal identifier here. This function would async return undefined - it's not going to yield anything.

Reply:
To add to what Taylor mentioned - the grammar explicitly enables yield as a keyword through a flag in the grammar.